### PR TITLE
AB#8758 Components serialization

### PR DESCRIPTION
**Pull request by sc-nelikostadinova**

### CODE REVIEW

**Comment by adoprog** [comment]
It would be good to also add some guidance on how to use them to README.md

File: componentspush.ps1 [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+foreach ($collection in $collections)
+{
+    Write-Host "Collection: " $collection.Name
+
+   foreach ($component in $collection.components)
+   {
+      Write-Host "  Component: " $component.Name
+
+      $componentUrl = "$endpoint/libraries/$libraryId/collections/$($collection.id)/components"
+      Invoke-WebRequest -Uri $componentUrl -Method Post -Body ($component | ConvertTo-Json -depth 100) -Headers $headers -ContentType "application/json"
```
Comment (jflheureux) [inline componentspush.ps1:23]: @adoprog I am getting errors when pushing serialized components to a new XM Cloud organization. Specifically if the serialized content contains collections that do not exist on the new XM Cloud organization:
```
Collection:  Static Components
  Component:  Promo
Invoke-WebRequest : {"statusCode":404,"error":"Not Found","message":"Could not find collection"}
At C:\projects\Sitecore.Demo.XmCloud.PlaySummit\componentspush.ps1:23 char:7
+       Invoke-WebRequest -Uri $componentUrl -Method Post -Body ($compo ...
+       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
    Version:  Name ( staged )
```

We should create the collections first if they do not already exist.

**Review by jflheureux — CHANGES_REQUESTED** [review]
I reviewed this PR locally while using the feature/8774-sitecore-components branch (with all the new serialized files)

File: docs/website.md [hunk 2]
```diff
@@ -50,6 +50,16 @@ The `\items` folder contains serialized Sitecore content items for this demo. Th
 
 See [Sitecore Content Serialization documentation](https://doc.sitecore.com/xp/en/developers/103/developer-tools/sitecore-content-serialization.html) for more information.
 
+### Serialized Sitecore Components
+
+The `\items\components` folder contains serialized Sitecore components, styles and datasources. The scrips that take care of the serialization are located at the root of the project.
+
+To push the serialized items to your Components library use the following command:
+.\componentspush.ps1 -libraryId _your_library_id_ -apiKey _your_api_key_
```
[the remark this answers is not in the document]
Comment (sc-nelikostadinova) [reply docs/website.md:58]: Will there be a better way to get that or should we explain the way to get it from the network tab? I felt like this was a temporary solution but I could be wrong.

[the remark this answers is not in the document]
Comment (jflheureux) [reply docs/website.md:58]: I do not know. We should ask Hande Bodart. Do you want to ask her or do you want me to ask?

**Comment by sc-nelikostadinova** [comment]
Closing because the fix for collections is in here: https://github.com/Sitecore/Sitecore.Demo.XmCloud.PlaySummit/pull/85

**Pull request closed without merging**